### DISCUSSION TITLE
Keycloak RabbitMQ/AMQP Listener SPI

### DISCUSSION THREAD

**Discussion by raduromaniuc** [keycloak/keycloak#14028] [Q&A]
### Description

Can Keycloak listen to events generated by a trusted RabbitMQ/AMQP broker? 

In another words, it would be very helpful if **Keycloak could act like an AMQP subscriber/consumer**, e.g. listening to event X and modifying its users' attributes based on the event information.

I have found the complement of this feature request:
https://github.com/aznamier/keycloak-event-listener-rabbitmq
But this makes Keycloak a publisher instead of a consumer.

Could this be possible?
Looking at the Keycloak documentation, there are a number of Providers listed, but none seems to help for this use-case. Any point in the right direction would be greatly appreciated

Thank you

### Motivation

Write now, we keep information in Keycloak about users as attributes, which are also present in tokens. These attributes sometimes change based on a RabbitMQ event. One event can change 100s of users' attributes: the current keycloak-admin API does not support bulk user updates, meaning we have to make lots of requests (lots of network overhead).

If instead Keycloak could directly listen to the AMQP event, we could directly modify the users' attributes, with minimal overhead.
But maybe there is too much of a security issue?

Thank you

**Comment by ergoz** [comment]
You can create consumer by your self that will work with keycloak admin api. Just for example you can use golang with ready to use package for keycloak and amqp (rabbitmq) package. It requires near 100 lines of code and can be flexible for your needs.
Sry for bad English, it is not native for me :)
